packages/mui-material/src/Autocomplete/Autocomplete.js

```diff
@@ -682,7 +682,7 @@ const Autocomplete = React.forwardRef(function Autocomplete(inProps, ref) {
   let autocompletePopper = null;
   if (groupedOptions.length > 0) {
     autocompletePopper = renderAutocompletePopperChildren(
-      <AutocompleteListbox as={ListboxSlot} {...listboxProps}>
+      <AutocompleteListbox as={ListboxSlot} ownerState={ownerState} {...listboxProps}>
```

Comment:
Reason for value of `ownerState` is undefined in styleOverrides is, absence of `ownerState` key in `listBoxprops`


`useSlot` hook doesn't attach ownerState if `elementType` is an html element , since elementType of ListBoxSlot is 'ul' not a React component (check [here](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Autocomplete/Autocomplete.js#L561) ), ownerState was missing in `listBoxprops`. 

https://github.com/mui/material-ui/blob/be68b0913c61eae68175c0b8d79c1b032fa383e4/packages/mui-utils/src/appendOwnerState/appendOwnerState.ts#L43-L46

Attaching ownerState explicity seems to fix the issue